File: mykoob/exceptions.py

class BadCredentialsException(Exception):
    """Exception raised when a user is not authenticated."""

    def __init__(self, message="Credentials invalid."):
        self.message = message
        super().__init__(self.message)
    
    def __str__(self):
        return f"BadCredentialsException: {self.message}"
    
class NotAuthenticatedError(Exception):
    """Exception raised when a user is not authenticated."""

    def __init__(self, message="User is not authenticated"):
        self.message = message
        super().__init__(self.message)

    def __str__(self):
        return f"NotAuthenticatedError: {self.message}"

class NoAccessTokenError(Exception):
    """Exception raised when no access token is available."""
    
    def __init__(self, message="No access token available"):
        self.message = message
        super().__init__(self.message)
        
    def __str__(self):
        return f"NoAccessTokenError: {self.message}"

class NoLessonsError(Exception):
    """Exception raised when no lessons retrieved by request"""

    def __init__(self, message="No lessons in request json!"):
        self.message = message
        super().__init__(self.message)
        
    def __str__(self) -> str:
        return f"NoLessonsError: {self.message}"


class BadResponseError(Exception):
    """Exception raised when bad response retrieved"""
    
    def __init__(self, message="Bad response. Maybe, some errors."):
        self.message = message
        super().__init__(self.message)
        
    def __str__(self):
        return f"BadResponseError: {self.message}"


class NoTokenError(Exception):
    """Exception raised when no token is available."""

    def __init__(self, message="Access token is missing or None"):
        self.message = message
        super().__init__(self.message)

    def __str__(self):
        return f"NoAccessTokenError: {self.message}"

class NoUserError(Exception):
    """Exception raised when no user is available."""
    def __init__(self, message="No user available"):
        self.message = message
        super().__init__(self.message)
        
    def __str__(self):
        return f"NoUserError: {self.message}"

File: javascript/src/exceptions.js
class BadCredentialsException extends Error {
    /**
     * Exception raised when a user is not authenticated due to bad credentials.
     * @param {string} message - The error message.
     */
    constructor(message = "Credentials invalid.") {
        super(message);
        this.name = "BadCredentialsException";
    }

    toString() {
        return `${this.name}: ${this.message}`;
    }
}

export class NoSessionError extends Error {
    /**
     * Exception raised when no session is provided.
     * @param {string} message - The error message.
     */

    constructor(message = "No session is provided") {
        super(message);
        this.name = "NoSessionError";
    }

    toString() {
        return `${this.name}: ${this.message}`;
    }
}

export class NotAuthenticatedError extends Error {
    /**
     * Exception raised when a user is not authenticated.
     * @param {string} message - The error message.
     */
    constructor(message = "User is not authenticated") {
        super(message);
        this.name = "NotAuthenticatedError";
    }

    toString() {
        return `${this.name}: ${this.message}`;
    }
}

class NoAccessTokenError extends Error {
    /**
     * Exception raised when no access token is available.
     * @param {string} message - The error message.
     */
    constructor(message = "No access token available") {
        super(message);
        this.name = "NoAccessTokenError";
    }

    toString() {
        return `${this.name}: ${this.message}`;
    }
}

class NoLessonsError extends Error {
    /**
     * Exception raised when no lessons are retrieved by request.
     * @param {string} message - The error message.
     */
    constructor(message = "No lessons in request json!") {
        super(message);
        this.name = "NoLessonsError";
    }

    toString() {
        return `${this.name}: ${this.message}`;
    }
}

class BadResponseError extends Error {
    /**
     * Exception raised when a bad response is retrieved.
     * @param {string} message - The error message.
     */
    constructor(message = "Bad response. Maybe, some errors.") {
        super(message);
        this.name = "BadResponseError";
    }

    toString() {
        return `${this.name}: ${this.message}`;
    }
}

export class NoTokenError extends Error {
    /**
     * Exception raised when no access token is available.
     * @param {string} message - The error message.
     */
    constructor(message = "Access token is missing or None") {
        super(message);
        this.name = "NoTokenError";
    }

    toString() {
        return `${this.name}: ${this.message}`;
    }
}

class NoUserError extends Error {
    /**
     * Exception raised when no user is available.
     * @param {string} message - The error message.
     */
    constructor(message = "No user available") {
        super(message);
        this.name = "NoUserError";
    }

    toString() {
        return `${this.name}: ${this.message}`;
    }
}

class ClassNotFoundError extends Error {
    /**
     * Exception raised when class is not found.
     * @param {string} message - The error message.
     */
    constructor(message = "Class not found") {
        super(message);
        this.name = "ClassNotFoundError";
    }

    toString() {
        return `${this.name}: ${this.message}`;
    }
}

